gather_gitops.sh

```diff
@@ -0,0 +1,177 @@
+#!/bin/bash
+BASE_COLLECTION_PATH="/must-gather"
+
+GITOPS_COLLECTION_PATH="$BASE_COLLECTION_PATH/cluster-gitops"
+gitops_folder="$GITOPS_COLLECTION_PATH/gitops"
+
+# terminal text output color
+green='\033[0;32m'!=
+color_reset='\033[0m' 
+
+function getNamespaces() {
+  local result
+  local namespaces
+  local default="openshift-gitops"
+  local clusterScopedInstances
+  clusterScopedInstances=$(oc get subs openshift-gitops-operator -n openshift-operators -o json | jq '.spec.config.env[]|select(.name=="ARGOCD_CLUSTER_CONFIG_NAMESPACES").value' | tr -d '",')
+  if [[ "$(oc get subs openshift-gitops-operator -n openshift-operators -o jsonpath='{.spec.config.env}')" == "" ]]; then
+    namespaces=${default}
+  elif [[ "${clusterScopedInstances}" != "" ]]; then
+    if [[ "$(oc get subs openshift-gitops-operator -n openshift-operators -o json | jq '.spec.config.env[]|select(.name=="DISABLE_DEFAULT_ARGOCD_INSTANCE").value')" != "false" ]]; then
+      namespaces+="${clusterScopedInstances} ${default}"
+    else
+      namespaces="${clusterScopedInstances}"
+    fi
+  else 
+    echo "no namespaces found"
+  fi
+
+  local argocdInstances
+  argocdInstances=$(oc get ArgoCD --all-namespaces -o jsonpath='{.items[*].metadata.namespace}')
+
+  local total
+  total="${namespaces} ${argocdInstances}"
+  
+  export NAMESPACES=$(echo "${total}" | tr ' ' '\n' | sort -u)
+}
+
+function main() {
+  
+  getNamespaces
+
+  echo -e "${green}Starting GitOps Operator must-gather script..."
+  mkdir -p "$gitops_folder"
+
+  echo -e "getting OpenShift Cluster Version..."
+  oc version > "${gitops_folder}"/oc-version.txt 2>&1
+
+  echo -e "getting GitOps Operator Subscription..."
+  oc get subs openshift-gitops-operator -n openshift-operators > "${gitops_folder}"/subscription.yaml 2>&1
+
+  echo -e "${NAMESPACES}"
+
+  for namespace in ${NAMESPACES}; do
+    mkdir -p ""${gitops_folder}"/namespace_"${namespace}"_resources"
```

Comment:
```suggestion
    mkdir -p "${gitops_folder}/namespace_${namespace}_resources"
```

The code is having some issues with missing character quotes, particularly the double quotes. e.g. `mkdir -p ""` is already a problem. Please revise all of the similar code-smells here, as I've noticed they all have similar issues.